prompt-service/src/unstract/prompt_service/controllers/answer_prompt.py

```
@@ -181,7 +181,9 @@ def prompt_processor() -> Any:
                 RunLevel.RUN,
                 "Unable to obtain LLM / embedding / vectorDB",
             )
-            raise APIError(message=msg)
+            # Preserve the status code from the SDK error
+            status_code = getattr(e, "status_code", None) or 500
+            raise APIError(message=msg, code=status_code)
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)